doc,win: clarify WSL support

joaocgreis opened the pull request:
Following the discussion in https://github.com/nodejs/node/issues/13471, this adds a note to `BUILDING.md` to clarify WSL support.

cc @nodejs/platform-windows @nodejs/build 

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->

doc, win

joaocgreis commented:
CI: https://ci.nodejs.org/view/All/job/node-test-pull-request/11411/

TimothyGu commented on BUILDING.md:
```diff
@@ -66,6 +66,12 @@ note1 - The gcc4.8-libs package needs to be installed, because node
+  the GNU/Linux build process and binaries should work. The community will
+  only address issues that reproduce on native GNU/Linux systems, issues that
+  only reproduce on WSL should be reported in the
+  [WSL issue tracker](https://github.com/Microsoft/WSL/issues).
```
It would be good to mention executing `node.exe` in WSL is not supported either.

refack replied:
```

Trott commented on BUILDING.md:
```diff
@@ -66,6 +66,12 @@ note1 - The gcc4.8-libs package needs to be installed, because node
 
+*Note*: The Windows Subsystem for Linux (WSL) is not directly supported, but
+  the GNU/Linux build process and binaries should work. The community will
+  only address issues that reproduce on native GNU/Linux systems, issues that
```
Nit: `systems, issues that` -> `systems. Issues that`

Trott commented on BUILDING.md:
```diff
@@ -66,6 +66,12 @@ note1 - The gcc4.8-libs package needs to be installed, because node
   In "Git bash" if you call the node shell alias (`node` without the `.exe`
   extension), `winpty` is used automatically.
 
+*Note*: The Windows Subsystem for Linux (WSL) is not directly supported, but
```
Nit: Perhaps remove `*Note*:` and let the paragraph stand on its own?

joaocgreis commented:
Addressed feedback, PTAL.

@TimothyGu @refack [Windows binaries should work in WSL](https://msdn.microsoft.com/en-us/commandline/wsl/interop#invoking-windows-binaries-from-wsl), so saying it's not supported is too strong. We don't want to work around bugs in WSL, but if it can provide a valid Windows environment it should work. I've left the "is not recommended", should be clear enough.

joaocgreis commented:
Landed in https://github.com/nodejs/node/commit/d8debd8448a1d325a564c42e5639655973b7641b

gibfahn commented:
@nodejs/build @nodejs/platform-windows I backported this to v8.x-staging, as I think the support statement is true for LTS lines as well. If anyone disagrees LMK.

bzoz commented:
SGTM
